api/src/tasks/generateNbTxsPerDay.ts

```diff
@@ -1,16 +1,22 @@
 import { addDays, isBefore, format } from "date-fns";
 import { prisma } from "../prisma";
-import { startDate } from "../utils";
+import { startDate, Result } from "../utils";
 
-export async function generateNbTxsPerDay() {
+export async function generateNbTxsPerDay(currentData: Result[]) {
   const endDate = new Date();
   let iteratorDate = startDate;
-  const result = [];
+  const result: Result[] | undefined = [];
+
+  const sortedData = currentData.sort((a, b) => {
```

Comment:
I don't think you need to sort it here as it's already sorted (latest date at the end of the array)